content/concept-history-sharding.md

```diff
@@ -0,0 +1,57 @@
+
+## XRP Ledger Network Data Integrity
+
+The history of all ledgers is shared by servers agreeing to keep particular ranges of historical ledgers. This makes it possible for servers to confirm that they have all the data they agreed to maintain, and trivializes the work required to produce proof trees or ledger deltas. Sharding also makes it possible to challenge servers to demonstrate they hold the data they claim to have, and verify their proofs.
+
+When a `rippled` server has too much recent history in the node store, the server has two options. It can discard the shard from recent storage, or it can move the shared from recent storage to historical storage and eject an existing shard. This decision is made following the purge logic for the shard feature.
+
+When a `rippled` server has more space than the configured allocation of shard store size, it assigns storage to begin storing shards or to store more shards. To increase the probability for an even distribution of the history of the network's ledgers, shards are selected randomly among the shards not currently held, with no preferences. Once chosen, the selected shard is fetched, validated, and committed to long term storage.
+
+## Shard Configuration
+Rippled can be configured to begin storing ledger history in shards by adding the `shard_db` section to the `rippled.cfg` file.
+
+### Shard Configuration Example
+The following snippet from an example `rippled.cfg` file shows the configuration fields for adding sharding to a `rippled` server:
+
+```
+# Begin Shard support
+[shard_db]
+type=NuDB
+path=/Users/{username}/.rippled/db/shards/nudb
+max_size_gb=50
+# End shard support
+```
+
+**Tip:** Ripple recommends setting the `type=NuDB` to use the NuDB for sharding, instead of the `RocksDB`.
+
+For more information, reference the `[shard_db]` example in the [rippled.cfg configuration example](https://github.com/ripple/rippled/blob/master/doc/rippled-example.cfg).
+
+***TODO: Question: What reason can we state for recommending NuDB over RocksDB?***
+
+***TODO: Question: Does this feature modify the NuDB header format with a network identifier and range of ledgers held, per David's article?***
+
+***TODO: Followup Question: If so, and if we should document the header and any relevant concerns here, what might those be?***
+
+### Sizing the Shard Store
+Determining a suitable value for the shard store will involve carefully considering different aspects of the process. Although redundant, it is entirely possible to hold full history in the node store and the shard store. Servers can shrink the range of ledgers they hold by dropping a database and then re-acquiring whatever ledgers they decide they do want to hold.
+
+The following concerns should each be considered in determining shard store sizing:
+
+- Each shard of the node store is 2^14 ledgers (at time of publishing, Q1 2018), which may not be optimal.
```

Comment:
Is it possible to replace this with more of a recipe for configuration to make it more actionable?  Or if not a general recipe, suggested scenarios (e.g. full-history, minimal node store + larger shard store, medium node store + medium shard store)?